units/provisioning_agent_deploy.go

```diff
@@ -150,6 +150,15 @@ func (j *agentDeployJob) Run(ctx context.Context) {
 	}
 
 	if stat.LastAttemptFailed() && stat.AllAttemptsFailed() && stat.Count >= agentPutRetries {
+		if j.host.Provider != evergreen.ProviderNameStatic {
+			externallyTerminated, err := CheckExternallyTerminatedHost(ctx, j.ID(), j.env, j.host)
+			if err != nil {
```

Comment:
You don't need to check `if err != nil`, since wrapping a nil error returns nil, and `j.AddError` does the right thing with nil.

Reply:
should be good now.